build.gradle.kts

```diff
@@ -348,7 +352,7 @@ tasks.register("createDmg") {
           "--type", "dmg",
           "--app-image", ext.get("appDirName") as String,
           "--name", project.name,
-          "--app-version", ext.get("appVersion") as String,
+          "--app-version", ext.get("appVersionShort") as String,
```

Comment:
It turns out that you did not need to change this line. But you do need to add this appVersionShort line to the createApp task at line 289. The failure occurs in createApp which is still getting its value from the sharedParameters. Once createApp succeeds, you can add the suffix to the .dmg name when running createDmg. Apparently jpackage only checks the version format when building the app (which needs to match Apple's version formatting). That way the app's Info.plist has the "real" version in a.b.c form and the Logisim-evolution-3.5.0-beta1.dmg file contains the suffix after the version.

I also realized in testing this, that jpackage will use the last setting of the parameter. So if it is set in the sharedParameters, it can be overridden by also including it in the list that is added in createApp. I'm not sure if that is a good idea. It simplifies our code; but at some time in the future, jpackage may decide to detect it has been set twice and complain. What do you think?

Reply:
I updated the macOS related tasks according to your notes. Please let me know if all is fine.

> jpackage may decide to detect it has been set twice and complain. What do you think?

I would not worry about that. Such detection would be of no sense really. In fact many CLI tools allow setting the same flag multiple times and the last one wins because that also simplifies constructing command line invocation strings. Anyway, we can worry once that become a real problem.